feat(logs): Adds link to log functionality

jonmcwest opened the pull request:
## Problem

Sharing a specific log entry wasn't easy

## Changes

<img width="1625" height="181" alt="image" src="https://github.com/user-attachments/assets/5a2a75ac-4529-4361-89f5-c7a5b92b249b" />

- Copyable link that scrolls directly to the log in question

## How did you test this code?

local dev

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

jonmcwest committed:
feat(logs): add highlighted log functionality as precursor to 'link to log' functionality

jonmcwest committed:
feat(logs): adds 'copy link to log' functionality

jonmcwest committed:
Update products/logs/frontend/LogsScene.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

frankh commented on products/logs/frontend/components/LogsTable/LogsTableRowActions.tsx:
```diff
@@ -1,16 +1,34 @@
+
+    const handleCopyLink = (): void => {
+        const url = new URL(window.location.href)
+        url.searchParams.set('highlightedLogId', log.uuid)
```
thinking about it i think we need to populate all filters here, it's very unlikely with our prod volume that the highlighted log would be visible after clearing the filters (we'll be showing 100 logs out of like 4+Million per minute)

jonmcwest committed:
fix(logs): preserve all filters when copying link to log

Previously cleared filters and only kept visible time range, making highlighted logs impossible to find in high-volume production environments. Now keeps all current filters and only adds highlightedLogId param.

frankh reviewed (approved):
LGTM 👍

jonmcwest committed:
fix(logs): fix scroll-to-highlighted after rebase
